other-samples/java-sample/src/main/java/com/microsoft/cognitiveservice/anomalydetection/Main.java

```diff
@@ -1,76 +1,111 @@
+import java.nio.charset.CharsetDecoder;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.List;
 
 public class Main {
     // **********************************************
     // *** Update or verify the following values. ***
     // **********************************************
 
     // Replace the subscriptionKey string value with your valid subscription key.
-    private static final String subscriptionKey = "<Subscription Key>";
-
-    // Choose which anomaly detection way you want to use and change the uriBase's second part
-    private static final String rootUrl = "https://westus2.api.cognitive.microsoft.com/anomalydetector/v1.0";
-    private static final String lastDetect = "/timeseries/last/detect";
-    private static final String entireDetect = "/timeseries/entire/detect";
-    private static final String uriBase = rootUrl + lastDetect;
-
-    public static void main(String[] args) throws FileNotFoundException  {
-        String resourceName = "/request-data.json";
-        InputStream is = Main.class.getResourceAsStream(resourceName);
-        if (is == null) {
-            throw new NullPointerException("Cannot find resource file " + resourceName);
+    private static final String OCP_APIM_SUBSCRIPTION_KEY = "Ocp-Apim-Subscription-Key";
+
+    private static final String CONTENT_TYPE = "Content-Type";
+
+    private static final String APPLICATION_JSON = "application/json";
+
+    private static final String SUBSCRIPTION_KEY = "<YOUR-SUBSCRIPTION-KEY>";
+
+    private static final String END_POINT = "<YOUR-ANOMALY-DETECTOR-END-POINT>";
+
+    private static final HttpPipelinePolicy AUTH_POLICY =
+            new AzureKeyCredentialPolicy(OCP_APIM_SUBSCRIPTION_KEY,
+                    new AzureKeyCredential(SUBSCRIPTION_KEY));
+
+    private static final AddHeadersPolicy ADD_HEADERS_POLICY =
+            new AddHeadersPolicy(new HttpHeaders().put(CONTENT_TYPE, APPLICATION_JSON));
+
+    private static final HttpPipeline HTTP_PIPELINE =
+            new HttpPipelineBuilder().policies(AUTH_POLICY, ADD_HEADERS_POLICY).build();
+
+    public static void main(String[] args) throws IOException {
+
+        AnomalyDetectorAsyncClient asyncClient = new AnomalyDetectorClientBuilder()
+                .endpoint(END_POINT).pipeline(HTTP_PIPELINE).buildAsyncClient();
+
+        JSONObject requestJson = new JSONObject(new JSONTokener((Main.class.getResourceAsStream("/request-data.json"))));
+        List<TimeSeriesPoint> timeSeriesPointList = new ArrayList<>();
+        JSONArray seriesArray = requestJson.getJSONArray("series");
+        for (int i = 0; i < seriesArray.length(); i++) {
+            JSONObject seriesObject = seriesArray.getJSONObject(i);
+            TimeSeriesPoint timeSeriesPoint = new TimeSeriesPoint()
+                    .setTimestamp(OffsetDateTime.ofInstant(Instant.parse(seriesObject.getString("timestamp")), ZoneOffset.UTC))
```

Comment:
This can be simplified to:
Instant.parse(seriesObject.getString("timestamp")).atOffset(ZoneOffset.UTC);

Rather than using another static method.

Reply:
According to the comment, it has been simplified to [ Instant.parse(series.timestamp()).atOffset(ZoneOffset.UTC) ].